### PR TITLE
Update pull_request_template.md to include accessibility compliance

### PR DESCRIPTION
**Pull request by assuntad23**
In an effort to enforce improvement of our accessibility compliance, we should include a checkbox for devs to remind them to check compliancy guidelines. 

I've linked just to the headings.md file, but I'm sure there are other guidelines we could point to that are simple and concise pointers. 

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]
- [X] not testable

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by assuntad23** [comment]
@hexylena @ElectronicBlueberry @dannon  In a meeting earlier today, @ElectronicBlueberry , @hujambo-dunia , and I talked about how to ensure that we are compliant with accessibility efforts. I suggested modifying the PR template as a reminder to devs to check with our latest compliancy guidelines. 

Are there other guidelines created by us (Galaxy folks) or that are simple and concise enough that we should include them here?

What do you guys think about this update?

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -9,5 +9,8 @@
 - [ ] Instructions for manual testing are as follows:
   1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]
 
+## Accessibility (for UI Components)
+- [ ] This Pull Request complies with Accessibility best practices defined [here](https://github.com/galaxyproject/galaxy/blob/dev/client/docs/headings.md)
```
Comment (hexylena) [inline .github/pull_request_template.md]:
```suggestion
- [ ] This Pull Request complies with [the Galaxy Accessibility best practices](https://github.com/galaxyproject/galaxy/blob/dev/client/docs/headings.md)
```

xref https://usability.yale.edu/web-accessibility/articles/links#link-text

**Comment by nsoranzo** [comment]
I like the idea of an additional checklist to make it clearer what a developer should check before opening a PR, but maybe make it more general instead of specific to accessibility, so it can include e.g.: "I have run `make format` (for Python changes)" and "I have run `make client-format` (for client changes)"

**Comment by dannon** [comment]
I definitely support adding more general guidance and linking up documentation talking about accessibility, formatting, linting, etc., (lots of which is in CONTRIBUTING.md, -- this actually came up recently when someone didn't know where exactly to look for that)

That said, the headings rule seems overly specific to call out in the PR template to me.  This is the kind of thing we really just need to enforce in linting and testing.  Linting is now available using vue-a11y, which will catch a lot of specific local-to-component a11y issues.  It can't catch the heading one, though, since it's testing components in isolation.  But that's where the work (on the ui/ux roadmap) towards automated axe testing comes in.

**Comment by assuntad23** [comment]
closing this because it does not seem like the approach that folks want to take, but I'm glad that it prompted some discussion.

**Pull request closed without merging**